components/feature/accounts/src/main/java/mozilla/components/feature/accounts/FxaWebChannelFeature.kt

```diff
@@ -299,5 +309,48 @@ class FxaWebChannelFeature(
                 }
             }
         }
+
+        private fun isCommunicationAllowed(serverConfig: ServerConfig, port: Port): Boolean {
+            val senderOrigin = port.senderUrl()
+            val expectedOrigin = serverConfig.contentUrl
+            return isCommunicationAllowed(senderOrigin, expectedOrigin)
+        }
+
+        @VisibleForTesting
+        internal fun isCommunicationAllowed(senderOrigin: String, expectedOrigin: String): Boolean {
+            if (!isSafeUrl(senderOrigin)) {
+                logger.error("$senderOrigin looks unsafe, aborting.")
+                return false
+            }
+
+            if (!isSameOrigin(senderOrigin, expectedOrigin)) {
+                logger.error("Host mismatch for WebChannel message. Expected: $expectedOrigin, got: $senderOrigin.")
+                return false
+            }
+            return true
+        }
+
+        /**
+         * Compares 2 URLs and returns true if they have the same origin,
+         * which means: same protocol, same host, same port.
+         */
+        private fun isSameOrigin(a: String, b: String): Boolean {
```

Comment:
Nice! Let's move this to a `String` extension function similar to https://github.com/mozilla-mobile/android-components/blob/master/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt#L105

Then we can also move the extensive test case over to StringUtilsTest. You can then also invoke this as `senderOrigin.isSameOriginAs(expectedOrigin)`.